### PR TITLE
Add/coverage api

### PR DESCRIPTION
**Pull request by hsiwkin**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,6 +5,7 @@
   "version": "1.0.25",
   "scripts": {
     "test": "grunt test",
+    "test:karma": "karma start tests/karma/single.conf.js",
```
Comment (TomaszLukawskiSam) [inline package.json]: use local karma package instead of global package
./node_modules/karma/bin/karma

File: package.json [hunk 2]
```diff
@@ -44,9 +45,10 @@
     "grunt-qunit-tap": "~2.0.0",
     "grunt-string-replace": "~0.2.4",
     "grunt-sync": "^0.6.2",
-    "karma": "^1.3.0",
+    "karma": "^4.4.1",
```
Comment (TomaszLukawskiSam) [inline package.json]: are you sure we need update karma version?

File: tests/karma/single.conf.js [hunk 3]
```diff
@@ -48,7 +48,11 @@ module.exports = function (config) {
 			{pattern: "tests/js/**/*.html", included: false, served: true, watch: true},
 			{pattern: "src/js/**/*.js", included: false, served: true, watch: true},
 			{pattern: "dist/**/*", included: false, served: true, watch: true},
-			"tests/karma/single.test.path.js",
+			{pattern: "libs/BezierCurve.js", included: false, served: true},
```
Comment (TomaszLukawskiSam) [inline tests/karma/single.conf.js]: Instead of modify this file we can update `all.conf.js`

File: tests/karma/single.conf.js [hunk 4]
```diff
@@ -85,11 +89,12 @@ module.exports = function (config) {
 			}
 		},
 
-		browsers: ["bigScreen"],
+		// browsers: ["bigScreen"],
+		browsers: ["ChromeHeadless"],
```
Comment (TomaszLukawskiSam) [inline tests/karma/single.conf.js]: I think a better solution is change `bigScreen.base` to `Chrome`

**Review by TomaszLukawskiSam — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (hsiwkin) [reply package.json]: I've restored the "old" version of Karma

File: tests/karma/single.conf.js (hunk 3 above)
Comment (hsiwkin) [reply tests/karma/single.conf.js]: done

File: tests/karma/all.conf.js [hunk 5]
```diff
@@ -50,6 +50,7 @@ module.exports = function (config) {
 			{pattern: "libs/**/*.js", included: false, served: true, watch: true},
 			{pattern: "dist/**/*", included: false, served: true, watch: true},
 			{pattern: "libs/path-to-regexp.js", included: false, served: true},
+			{pattern: "libs/BezierCurve.js", included: false, served: true},
 			"tests/karma/testPaths.js",
```
Comment (TomaszLukawskiSam) [inline tests/karma/all.conf.js:55]: I think that file`{pattern: "some-script.js", included: false, served: true},`  is also needed.

File: tests/karma/uicomponents.mobile.conf.js [hunk 6]
```diff
@@ -70,7 +67,8 @@ module.exports = function (config) {
 			}
 		},
 
-		browsers: ["bigScreen"],
+		// browsers: ["bigScreen"],
+		browsers: ["ChromeHeadless"],
```
Comment (TomaszLukawskiSam) [inline tests/karma/uicomponents.mobile.conf.js]: What size of screen will be after your change?

File: tests/karma/all.conf.js (hunk 5 above)
Comment (hsiwkin) [reply tests/karma/all.conf.js:55]: right, done

File: tests/karma/all.conf.js [hunk 7]
```diff
@@ -82,16 +84,15 @@ module.exports = function (config) {
 		// available browser launchers: https://npmjs.org/browse/keyword/karma-launcher
 		customLaunchers: {
 			bigScreen: {
-				base: "Chrome",
-				flags: ["--window-size=800,800"]
```
Comment (TomaszLukawskiSam) [inline tests/karma/all.conf.js:86]: Why do you remove "--widndow-size", is it not supported?

**Review by TomaszLukawskiSam — CHANGES_REQUESTED** [review]

File: tests/karma/all.conf.js (hunk 7 above)
Comment (hsiwkin) [reply tests/karma/all.conf.js:86]: I thought it isn't but apparently it is supported: 
https://stackoverflow.com/questions/43541925/how-to-set-the-browser-window-size-when-using-google-chrome-headless
https://peter.sh/experiments/chromium-command-line-switches/#window-size

Comment (hsiwkin) [reply tests/karma/all.conf.js:86]: It's back

**Review by TomaszLukawskiSam — APPROVED** [review]
Looks good to me.

**Pull request merged**